test/unit/libraries/Resealer.t.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+import {Resealer} from "contracts/libraries/Resealer.sol";
+import {IResealManager} from "contracts/interfaces/IResealManager.sol";
+import {UnitTest} from "test/utils/unit-test.sol";
+
+contract ResealerTest is UnitTest {
+    using Resealer for Resealer.Context;
+
+    Resealer.Context ctx;
+
+    address resealManager = address(0x1);
+    address resealCommittee = address(0x2);
+
+    function setUp() external {
+        ctx.resealManager = IResealManager(resealManager);
+        ctx.resealCommittee = resealCommittee;
+    }
+
+    function test_setResealManager_HappyPath(address newResealManager) external {
+        vm.assume(newResealManager != address(ctx.resealManager) && newResealManager != address(0));
+
+        vm.expectEmit();
+        emit Resealer.ResealManagerSet(newResealManager);
+        ctx.setResealManager(newResealManager);
+
+        assertEq(address(ctx.resealManager), newResealManager);
+    }
+
+    function test_setResealManager_RevertOn_InvalidResealManager() external {
+        vm.expectRevert(abi.encodeWithSelector(Resealer.InvalidResealManager.selector, address(ctx.resealManager)));
+        ctx.setResealManager(address(ctx.resealManager));
```

Comment:
`expectRevert()` doesn't work properly with the internal libraries methods, so a more proper approach is testing it via external call as in `test_checkCallerIsResealCommittee_HappyPath `.